arch/xtensa/src/esp32s3/esp32s3_gpio.c

```diff
@@ -141,25 +202,37 @@ int esp32s3_configgpio(int pin, gpio_pinattr_t attr)
  * Name: esp32s3_gpio_matrix_in
  *
  * Description:
- *   Set gpio input to a signal
- *   NOTE: one gpio can input to several signals
- *   If gpio == 0x3c, cancel input to the signal, input 0 to signal.
- *   If gpio == 0x3a, input nothing to signal.
- *   If gpio == 0x38, cancel input to the signal, input 1 to signal.
+ *   Set GPIO input to a signal.
+ *   NOTE: one GPIO can input to several signals.
+ *
+ * Input Parameters:
+ *   pin           - GPIO pin to be configured.
+ *                   - If pin == 0x3c, cancel input to the signal, input 0
+ *                     to signal.
+ *                   - If pin == 0x3a, input nothing to signal.
+ *                   - If pin == 0x38, cancel input to the signal, input 1
+ *                     to signal.
+ *   signal_idx    - Signal index.
+ *   inv           - Flag indicating whether the signal is inverted.
+ *
+ * Returned Value:
+ *   None.
  *
  ****************************************************************************/
 
-void esp32s3_gpio_matrix_in(uint32_t gpio, uint32_t signal_idx, bool inv)
+void esp32s3_gpio_matrix_in(uint32_t pin, uint32_t signal_idx, bool inv)
 {
   uint32_t regaddr = GPIO_FUNC0_IN_SEL_CFG_REG + (signal_idx * 4);
-  uint32_t regval = (gpio << GPIO_FUNC0_IN_SEL_S);
+  uint32_t regval = (pin << GPIO_FUNC0_IN_SEL_S);
+
+  DEBUGASSERT(is_valid_gpio(pin));
 
   if (inv)
     {
       regval |= GPIO_FUNC0_IN_INV_SEL;
     }
 
-  if (gpio != 0x3a)
+  if (pin != 0x3a)
```

Comment:
`0x3a == 58` and `ESP32S3_NPINS` is `49`, so this condition seems to be never hit. Should `signal_idx` be checked instead of `pin`?

Reply:
`pin` is the one to compared against **0x3a**, but the issue here is that `esp32s3_gpio_matrix_in` should not check if `pin` is within the valid range of GPIOs.

Nice catch, thanks again!

Reply:
What about the other values?
```
 *   pin           - GPIO pin to be configured.
 *                   - If pin == 0x3c, cancel input to the signal, input 0
 *                     to signal.
 *                   - If pin == 0x3a, input nothing to signal.
 *                   - If pin == 0x38, cancel input to the signal, input 1
```
should those be checked?

Reply:
The **0x3c** and **0x38** values will be handled by the GPIO Matrix peripheral, so no need to perform extra verification here.